src/main/java/io/leonis/algieba/spatial/AggregatedPotentialField.java

```diff
@@ -0,0 +1,34 @@
+package io.leonis.algieba.spatial;
+
+import java.util.Set;
+import lombok.Value;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+/**
+ * The Class AggregatedPotentialField.
+ *
+ * This class represents an aggregation of multiple {@link PotentialField}.
+ *
+ * @author Rimon Oz
+ */
+@Value
```

Comment:
We might want to reconsider using `@Value` everywhere. since `AggregatedPotentialField` shouldn't expose `potentialFields`. a `Getter`, `AllArgsConstructor` and final notation might be better suited here.